--- a/xbmc/interfaces/Builtins.cpp	2013-08-01 10:42:32.721925019 -0400
+++ b/xbmc/interfaces/Builtins.cpp	2013-07-30 01:35:09.959541000 -0400
@@ -463,16 +463,25 @@ int CBuiltins::Execute(const CStdString&
   else if (execute.Equals("resolution"))
   {
     RESOLUTION res = RES_PAL_4x3;
-    if (parameter.Equals("pal")) res = RES_PAL_4x3;
-    else if (parameter.Equals("pal16x9")) res = RES_PAL_16x9;
-    else if (parameter.Equals("ntsc")) res = RES_NTSC_4x3;
-    else if (parameter.Equals("ntsc16x9")) res = RES_NTSC_16x9;
-    else if (parameter.Equals("720p")) res = RES_HDTV_720p;
-    else if (parameter.Equals("720pSBS")) res = RES_HDTV_720pSBS;
-    else if (parameter.Equals("720pTB")) res = RES_HDTV_720pTB;
-    else if (parameter.Equals("1080pSBS")) res = RES_HDTV_1080pSBS;
-    else if (parameter.Equals("1080pTB")) res = RES_HDTV_1080pTB;
-    else if (parameter.Equals("1080i")) res = RES_HDTV_1080i;
+    if (parameter.Equals("pal"))            res = RES_PAL_4x3;
+    else if (parameter.Equals("pal16x9"))   res = RES_PAL_16x9;
+    else if (parameter.Equals("ntsc"))      res = RES_NTSC_4x3;
+    else if (parameter.Equals("ntsc16x9"))  res = RES_NTSC_16x9;
+    else if (parameter.Equals("720p"))      res = RES_HDTV_720p;
+    else if (parameter.Equals("720pSBS"))   res = RES_HDTV_720pSBS;
+    else if (parameter.Equals("720pTB"))    res = RES_HDTV_720pTB;
+    else if (parameter.Equals("1080pSBS"))  res = RES_HDTV_1080pSBS;
+    else if (parameter.Equals("1080pTB"))   res = RES_HDTV_1080pTB;
+    else if (parameter.Equals("1080i"))     res = RES_HDTV_1080i;
+    else
+    {
+      // format: SWWWWWHHHHHRRR.RRRRRP, where,
+      //  S = screen, W = width, H = height, R = refresh, P = interlace
+      res = g_guiSettings.GetResFromString(parameter);
+      const char *strMode = g_settings.m_ResInfo[res].strMode.c_str();
+      CLog::Log(LOGERROR,"resolution:res(%d), parameter(%s), strMode(%s)",
+        res, parameter.c_str(), strMode);
+    }
     if (g_graphicsContext.IsValidResolution(res))
     {
       g_guiSettings.SetResolution(res);

